### PR TITLE
Header bug fix

### PR DESCRIPTION
**Pull request by metascroy**
Summary: A last minute change created a compile error on the header.  This fixes the issue.

Differential Revision: D64370707

### CODE REVIEW

**Comment by malfet** [comment]
I'm unsure what it fixes to be frank, perhaps better PR description would help clarify what it is trying to fix...

File: torchao/experimental/ops/linear_8bit_act_xbit_weight/packed_weights_header.h
Comment (malfet) [inline torchao/experimental/ops/linear_8bit_act_xbit_weight/packed_weights_header.h:21]: What is wrong with this check?

File: torchao/experimental/ops/packed_weights_header.h
Comment (malfet) [inline torchao/experimental/ops/packed_weights_header.h:13]: Why this must be a signed type? `unsigned int` sounds like a better idea. Also `enum` inside the namespace is oxymoron, please use `enum class`
```suggestion
enum class PackedWeightsFormat : uint32_t {
```

File: torchao/experimental/ops/packed_weights_header.h
Comment (malfet) [inline torchao/experimental/ops/packed_weights_header.h:34]: Why not use sizeof(XYZ)?
```suggestion
    return sizeof(PackedWeightsHeader);
```

File: torchao/experimental/ops/packed_weights_header.h
Comment (malfet) [inline torchao/experimental/ops/packed_weights_header.h:38]: In C++ better used C++ cast-style ops
```suggestion
    auto header = reinterpret_cast<int*>(packed_weights);
```

File: torchao/experimental/ops/linear_8bit_act_xbit_weight/packed_weights_header.h
Comment (metascroy) [reply torchao/experimental/ops/linear_8bit_act_xbit_weight/packed_weights_header.h:21]: Before it existed to make sure version could be packed into a char (since both version/weight_nbit were packed into an unsigned short).  But the header now has more space, and version is now an int, so this bound check is no longer needed.

File: torchao/experimental/ops/packed_weights_header.h
Comment (metascroy) [reply torchao/experimental/ops/packed_weights_header.h:34]: I thought this made it clearer that the header is 64 bytes at a glance.

**Review by digantdesai — APPROVED** [review]

**Comment by digantdesai** [comment]
Address Nikita's comments before landing, stamping to unblock if you have already done so.

**Pull request merged**
